provider/aws/ec2volumes.go

```diff
@@ -0,0 +1,190 @@
+package aws
+
+import (
+	"encoding/json"
+	"finala/storage"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	log "github.com/sirupsen/logrus"
+)
+
+// EC2VolumeClientDescreptor is an interface defining the AWS EC2
+type EC2VolumeClientDescreptor interface {
+	DescribeVolumes(input *ec2.DescribeVolumesInput) (*ec2.DescribeVolumesOutput, error)
+}
+
+// EC2VolumeManager describe EBS manager
+type EC2VolumeManager struct {
+	client             EC2VolumeClientDescreptor
+	storage            storage.Storage
+	pricingClient      *PricingManager
+	region             string
+	namespace          string
+	servicePricingCode string
+}
+
+// DetectedAWSEC2Volume define the detected volume data
+type DetectedAWSEC2Volume struct {
+	Region        string
+	ID            string
+	Type          string
+	Size          int64
+	PricePerMonth float64 `gorm:"type:DOUBLE`
+	Tags          string  `gorm:"type:TEXT" json:"-"`
+}
+
+// TableName will set the table name to storage interface
+func (DetectedAWSEC2Volume) TableName() string {
+	return "aws_ec2_volume"
+}
+
+// NewVolumesanager implements AWS GO SDK
+func NewVolumesanager(client EC2VolumeClientDescreptor, st storage.Storage, pricing *PricingManager, region string) *EC2VolumeManager {
```

Comment:
I think you meant NewVolumesanager -> NewVolumesManager

Reply:
Fixed